### PR TITLE
SNOW-1462359/SNOW-1462481: Add Series.str and Series.dt to docs

### PR DESCRIPTION
**Pull request by sfc-gh-joshi**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1462359 and SNOW-1462481

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

This PR adds documentation for Series.str.* and Series.dt.* to Snowpark pandas. This is significant for two reasons:
1. This is the first time we are documenting frontend classes that are defined exclusively in upstream OSS modin, as we no longer have vendored copies of StringMethods and DatetimeProperties.
2. Series.str and Series.dt are technically separate classes, and a large amount of customization is necessary to render their methods under the `Series.str`/`dt` name rather than their respective accessor objects.

The changes here are based loosely off those made by native pandas for the same purpose: https://github.com/pandas-dev/pandas/blob/bbe0e531383358b44e94131482e122bda43b33d7/doc/source/conf.py#L484-L485

I could not get the changes of upstream pandas to work directly out of the box, though I am not entirely sure as to why. It could be a consequence of these classes being external to Snowpark, or just the fact that vanilla pandas has an `Accessor` class wrapping the `Series.str` and `Series.dt` properties that modin lacks.

See comments in `docs/source/conf.py` for a detailed explanation of what each class does.

### CODE REVIEW

File: docs/source/modin/supported/series_str_supported.rst [hunk 1]
```diff
@@ -117,9 +117,6 @@ the method in the left column.
 +-----------------------------+---------------------------------+----------------------------------------------------+
 | ``slice_replace``           | N                               |                                                    |
 +-----------------------------+---------------------------------+----------------------------------------------------+
-| ``replace``                 | P                               |  ``N`` if `pat` is non-string, `repl` is           |
```
Comment (sfc-gh-joshi) [inline docs/source/modin/supported/series_str_supported.rst:120]: This entry was redundant (it appears also on line 101, in the correct alphabetical location).

File: src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def split():
 
         Returns
         -------
-        Series, Index, DataFrame or MultiIndex
+        :class:`~snowflake.snowpark.modin.pandas.Series`, Index, :class:`~snowflake.snowpark.modin.pandas.DataFrame` or MultiIndex
```
Comment (sfc-gh-joshi) [inline src/snowflake/snowpark/modin/plugin/docstrings/series_utils.py:41]: This fixes a warning about `DataFrame` being ambiguous.

**Comment by sfc-gh-joshi** [comment]
FYI @sfc-gh-vbudati we may be able to stop replacing "snowflake.snowpark.pandas" in jinja templates moving forward. Setting `currentmodule` to `modin.pandas` within the rst files should be sufficient, as this PR demonstrates that the docstring plugin is correctly applied to upstream modin classes.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM, thanks!

**Review by sfc-gh-vbudati — APPROVED** [review]
This is awesome! Thanks for figuring this out, Jonathan!

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**